### DISCUSSION TITLE
Does SWC work with TypeScript project references?

### DISCUSSION THREAD

**Discussion by osdiab** [swc-project/swc#2156] [Q&A] [answered]
Can't seem to find any mention of it in the docs or issues, but I use those (the `composite` flag in `tsconfig.json`) heavily alongside `yarn` workspaces in my own repo - interested in knowing to decide if I should use SWC on my next project :)

**Comment by kdy1** [comment, marked as answer]
No, it's not supported.

**Reply by osdiab** [reply]
is it out of scope for SWC in general?

**Reply by kdy1** [reply]
No, it's in the scope, but I'm quite busy.

**Reply by osdiab** [reply]
cool, I'll be looking forward to that whenever it comes out!

**Reply by transitive-bullshit** [reply]
Oooooh also really looking forward to this 💯

**Reply by steinerkelvin** [reply]
Is there an issue tracking this?

**Reply by hasanayan** [reply]
Is this still the case? @kdy1

**Reply by yuntian001** [reply]
I need it

**Comment by ssalka** [comment]
Any chance this might be coming soon? 🙂 

I am working on deploying a server to Vercel, and since Vercel runs your build for you, I had to move TS into my package's `dependencies` list. However the build still fails because none of the `@types` packages are found (I'm running `npm install --production` so devDependencies are excluded). I thought I could get around this by adding `--noCheck`, but there's an [open TypeScript issue](https://github.com/microsoft/TypeScript/issues/58336) around the use of `--noCheck` with `-b`, so my options are to either move all `@types` packages into my dependencies, which is universally not recommended, or use an alternative compiler until the above issue is resolved.

SWC seems like the best alternative, but I'm using project references, so that's a no-go as well. I think for now I will have to remove the `--production` from my install step, which generally will be bad for install/build times.
